dart/lib/src/http_client/failed_request_client.dart

```diff
@@ -183,6 +186,19 @@ class FailedRequestClient extends BaseClient {
       throwable: throwableMechanism,
       request: sentryRequest,
     );
+
+    if (response != null) {
+      event.contexts.response = SentryResponse(
+        headers: sendDefaultPii ? response.headers : null,
+        redirected: response.isRedirect,
+        // Body not captured yet.
+        // See https://github.com/getsentry/sentry-dart/pull/1093#issuecomment-1293056244
+        body: null,
+        statusCode: response.statusCode,
+        status: response.reasonPhrase,
+      );
+    }
```

Comment:
Let's only set the values that are defined here https://github.com/getsentry/rfcs/blob/main/text/0022-response-context.md#proposal-namely-option-2
Let's add `@Deprecated` to the other fields, such as `url`, `redirected`, `body`, `status`, and `other`.
Let's add `body_size` here and also `SentryRequest`.

Reply:
We can also remove `@experimental` from the usage of SentryRequest, maybe we can even remove such fields (the ones mentioned above) since its still marked as experimental, wdyt?